homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,486 @@
+    """Defines a FRITZ!Box Tools PortForward switch."""
+
+    def __init__(self, fritzbox_tools, device_friendly_name, dict_of_deflection):
+        """Init Fritxbox Deflection class."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self.dict_of_deflection = dict_of_deflection
+        self._attributes = {}
+        self.id = int(self.dict_of_deflection["DeflectionId"])
+
+        switch_info = SwitchInfo(
+            description=f"Call deflection {self.id}",
+            friendly_name=device_friendly_name,
+            icon="mdi:phone-forward",
+            type=SWITCH_TYPE_DEFLECTION,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self.fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self):
+        """Fetch updates."""
+
+        try:
+            resp = await self.hass.async_add_executor_job(
+                lambda: self.fritzbox_tools.connection.call_action(
+                    "X_AVM-DE_OnTel:1", "GetDeflections"
+                )
+            )
+            self.dict_of_deflection = xmltodict.parse(resp["NewDeflectionList"])[
+                "List"
+            ]["Item"]
+            if isinstance(self.dict_of_deflection, list):
+                self.dict_of_deflection = self.dict_of_deflection[self.id]
+
+            _LOGGER.debug(
+                "Specific %s response: NewDeflectionList=%s",
+                SWITCH_TYPE_DEFLECTION,
+                self.dict_of_deflection,
+            )
+
+            self._attr_is_on = self.dict_of_deflection["Enable"] == "1"
+            self._is_available = True
+
+            self._attributes["Type"] = self.dict_of_deflection["Type"]
+            self._attributes["Number"] = self.dict_of_deflection["Number"]
+            self._attributes["DeflectionToNumber"] = self.dict_of_deflection[
+                "DeflectionToNumber"
+            ]
+            # Return mode sample: "eImmediately"
+            self._attributes["Mode"] = self.dict_of_deflection["Mode"][1:]
+            self._attributes["Outgoing"] = self.dict_of_deflection["Outgoing"]
+            self._attributes["PhonebookID"] = self.dict_of_deflection["PhonebookID"]
+
+        except FritzConnectionException:
+            _LOGGER.error(
+                "Authorization Error: Please check the provided credentials and verify that you can log into the web interface"
+            )
+            self._is_available = False
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error(
+                "Could not get state of %s", SWITCH_TYPE_DEFLECTION, exc_info=True
+            )
+            self._is_available = False
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle deflection switch."""
+        self.hass.async_add_executor_job(
+            lambda: self.fritzbox_tools.connection.call_action(
```

Comment:
Use functools.partial instead of the lambda

Reply:
No more needed with latest cleanup